cogs/voting.py

```diff
@@ -183,6 +183,28 @@ async def validate_referendum(self, context, author):
         await context.send(output_str)
         return valid
 
+# Error handling #
+
+    async def dm_error(self, context, error):
+        if isinstance(error, commands.errors.PrivateMessageOnly):
+            await context.send('This command is DM only, please try again in a private message to me.')
+            return True
+
+    @register.error
+    async def register_error(self, context, error):
+        if not await self.dm_error(context, error):
+            if isinstance(error, commands.errors.MissingRequiredArgument):
+                await context.send('Must supply a student number. Usage: {helpers.PREFIX}register <STUDENT NUMBER>')
+
+    @submit.error
+    async def submit_error(self, context, error):
+        if not await self.dm_error(context, error):
+            if isinstance(error, commands.errors.MissingRequiredArgument):
+                await context.send('Must supply the code. Usage: {helpers.PREFIX}register <STUDENT NUMBER>')
```

Comment:
I'm not 100% on how these error handlers work, but isn't the `if not code` check in the submit function superseded by this, so it can go?

Reply:
And if so, would it be possible to get this message to mention that they need to get the code from the election call?